sdk/identity/identity/samples/AzureIdentityExamples.md

```diff
@@ -431,6 +453,255 @@ function main() {
+    return {
+      token: result.accessToken,
+      expiresOnTimestamp: result.expiresOn.getTime()
+    }
+  }
+}
+```
+
+Users could then use the `ConfidentialClientApplicationCredential` to authenticate a `SecretClient` from `@azure/keyvault-secrets` with an MSAL `ConfidentialClientApplication`:
+
+```ts
+import { SecretClient } from "@azure/keyvault-secrets";
+import * as msalNode from "@azure/msal-node";
+
+async function main() {
+  const confidentialClient = new msalNode.ConfidentialClientApplication({
+    // MSAL Configuration
+  });
+
+  const client = new SecretClient("https://myvault.vault.azure.net/", new ConfidentialClientCredential(confidentialClient));
+}
+```
+
+#### Authenticating with the On Behalf Of Flow
+
+Currently the `@azure/identity` library doesn't provide a credential type for clients which need to authenticate via the On Behalf Of flow. While future support for this is planned, users currently requiring this will have to implement their own `TokenCredential` class.
+
+In this example the `OnBehalfOfCredential` accepts a client Id, client secret, and a user's access token. It then creates an instance of `ConfidentialClientApplication` from MSAL to obtain an OBO token which can be used to authenticate client requests.
+
+```ts
+import { TokenCredential, AccessToken } from "@azure/core-http";
+import * as msalNode from "@azure/msal-node";
+
+class OnBehalfOfCredential implements TokenCredential {
+  private confidentialApp: msalNode.ConfidentialClientApplication;
+
+  constructor(private clientId: string, private clientSecret: string, private userAccessToken: string) {
+    this.confidentialApp = new msalNode.ConfidentialClientApplication({
+      auth: {
+        clientId,
+        clientSecret
+      }
+    });
+  }
+  async getToken(scopes: string | string[]): Promise<AccessToken> {
+    const result = await this.confidentialApp.acquireTokenOnBehalfOf({
+      scopes: Array.isArray(scopes) ? scopes : [scopes],
+      oboAssertion: this.userAccessToken
+    });
+    return {
+      token: result.accessToken,
+      expiresOnTimestamp: result.expiresOn.getTime()
+    };
+  }
+}
+```
+
+The following example shows an how the `OnBehalfOfCredential` could be used to authenticate a `SecretClient`.
+
+```ts
+import { SecretClient } from "@azure/keyvault-secrets";
+
+async function main() {
+  const oboCredential = new OnBehalfOfCredential(clientId, clientSecret, userAccessToken);
+
+  const client = new SecretClient("https://myvault.vault.azure.net/", oboCredential);
+}
+```
+### Authenticating with Key Vault Certificates
+
+Azure Key Vault allows users to create certificates that can be used to authenticate Azure SDK clients.
+
+Certificates can be created through different means. You may follow any of these approaches:
+
+- [Quickstart: Set and retrieve a certificate from Azure Key Vault using the Azure portal](https://docs.microsoft.com/azure/key-vault/certificates/quick-create-portal).
```

Comment:
Maybe we should just link to a single one? Like the CLI for example? Users can then navigate the many different quickstarts from the sidebar menu